src/machine/machine_atsamd51.go

```diff
@@ -335,7 +335,7 @@ func (p Pin) SetInterrupt(change PinChange, callback func(Pin)) error {
 	pinCallbacks[extint] = callback
 	interruptPins[extint] = p
 
-	if (sam.EIC.CTRLA.Get() & 0x02) == 0 {
+	if (sam.EIC.CTRLA.Get() & sam.EIC_CTRLA_ENABLE) == 0 {
```

Comment:
You could use the `volatile` package `HasBits()` function here:

```
if !sam.EIC.CTRLA.HasBits(sam.EIC_CTRLA_ENABLE) {
```

Reply:
The source code has been modified.
It's better to use HasBits().